Update the generic rule for packages to use namespaces

jeremycline opened the pull request:
This depends on a change in the Fedora Processor changing to accept a
new kwarg, ``namespace``, that accounts for the namespace a package is
in when performing ``msg2packages``.

fixes #176

Note: The tests currently don't pass (since there needs to be a fix in https://github.com/fedora-infra/fedmsg_meta_fedora_infrastructure) and this shouldn't be merged yet.

jeremycline committed:
Modify the pkgdb2 call to include namespace information

This also adds unit tests for modified functions.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Update the generic rule for packages to use namespaces

This depends on a change in the Fedora Processor changing to accept a
new kwarg, ``namespace``, that accounts for the namespace a package is
in when performing ``msg2packages``.

fixes #176

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

bowlofeggs commented on fmn/rules/generic.py:
```diff
@@ -143,7 +151,11 @@ def user_package_watch_filter(config, message, fasnick=None, *args, **kw):
+            namespaced_msg_packages = fmn.rules.utils.msg2packages(
+                message, namespace=namespace, **config)
+            if packages.intersection(namespaced_msg_packages):
+                return True
```
This code block seem duplicated with the other two changes in this file. Perhaps make a shared function?

bowlofeggs commented on fmn/tests/rules/test_utils.py:
```diff
@@ -0,0 +1,127 @@
+                'erlang-p1_pgsql',
+                'erlang-p1_utils',
+                'erlang-p1_xmlrpc',
+                'erlang-p1_zlib',
```
Oh yeah, some erlang!

bowlofeggs reviewed (approved):
Just one suggestion.

jeremycline committed:
Factor out common code of generic rules into a function

This also removes an incorrect short-circuit in the function that is no
longer safe due to namespaces.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Update the cache key to include the namespace

Without counting the namespace as part of the call to msg2packages, we
can return incorrect cache results.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Explicitly require fedmsg_meta >= 0.18

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline commented:
@bowlofeggs I've made a few significant changes so it'd be good to have a re-review.

bowlofeggs commented on fmn/tests/rules/test_generic.py:
```diff
@@ -0,0 +1,176 @@
+from __future__ import unicode_literals
+
+from fedmsg.meta import make_processors
+import unittest
```
For PEP-8 you might make these two imports be in different groups.
